static/app/components/modals/addDashboardWidgetModal.tsx

```diff
@@ -340,10 +350,18 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
       fromDiscover,
       onUpdateWidget,
       widget: previousWidget,
+      start,
+      end,
+      statsPeriod,
     } = this.props;
     const state = this.state;
     const errors = state.errors;
 
+    const querySelection = statsPeriod
```

Comment:
can we do this instead of casting global selection below?
```suggestion
    const querySelection: GlobalSelection = statsPeriod
```

Reply:
hmm in this case, should we also be persisting environment and projects selection to the dashboard? If it gets too complicated, I think we can just use the default selection as the first pass?

Reply:
Updated this! Had to make a slight tweak to how `querySelection` is constructed to please typescript but functionally it works the same.